sphinx/highlighting.py

```diff
@@ -50,6 +52,20 @@
 _LATEX_ADD_STYLES = r'''
 \renewcommand\PYGZsq{\textquotesingle}
 '''
+# fix extra space between lines when Pygments highlighting uses \fcolorbox
+# add a {..} to limit \fboxsep scope, and force \fcolorbox use correct value
+# cf pygments #1708 which makes this unneeded for Pygments > 2.7.4
+_LATEX_ADD_STYLES_FIXPYG = r'''
+\makeatletter
+% fix for Pygments <= 2.7.4
+\let\spx@original@fcolorbox\fcolorbox
```

Comment:
This means `\fcolorbox` (from color/xcolor package) must exist when sphinxhighlight.sty is loaded by sphinx.sty. But this is the case as color/xcolor is loaded very early by sphinx.sty, before handling options. So we are safe here. (this is true also on 3.x even though sphinxhighlight.sty is loaded earlier, but I feel now latex changes must only go to master). One could add some `\AtBeginDocument` to be extra safe if some package modifies `\fcolorbox` but I feel this is unnecessary. Besides in future, this patch will not be used when people have updated their Pygments version.